lib/aur-chroot

```diff
@@ -184,88 +210,147 @@ elif [[ ! -f $makepkg_conf ]]; then
     exit 2
 fi
 
+bindmounts_ro+=("$pacman_conf" "$makepkg_conf" /etc/pacman.d/mirrorlist)
+
 # Print paths to container and used makepkg/pacman paths. This does
 # not require a priorly created container.
 if (( status )); then
     printf 'chroot:%s\npacman:%s\nmakepkg:%s\n' "$directory" "$pacman_conf" "$makepkg_conf"
     exit 0
 fi
 
-# Custom elevation command (#1024)
-unset auth_args
-AUR_PACMAN_AUTH=${AUR_PACMAN_AUTH:-sudo}
-
-case $AUR_PACMAN_AUTH in
-    sudo) auth_args+=('--preserve-env=GNUPGHOME,SSH_AUTH_SOCK,PKGDEST') ;;
-esac
-
-# bind mount file:// paths to container (#461)
-# required for update/build steps
+is_multilib=0
+cache_dirs=()
 while read -r key _ value; do
     case $key=$value in
         Server=file://*)
+            # bind mount file:// paths to container (#461)
+            # required for update/build steps
             bindmounts_rw+=("${value#file://}") ;;
+        CacheDir=*)
+            cache_dirs+=("${value}")
+            bindmounts_rw+=("${value}") ;;
+        \[multilib\]=)
+            is_multilib=1 ;;
     esac
 done < <(pacman-conf --config "$pacman_conf")
 wait "$!"
 
+cleanup() {
+    rm "$directory"/root/aur-chroot--helper
+}
+
+trap cleanup EXIT
+
 # create new container, required for update/build steps
-if (( create )); then
+if [[ ! -d $directory/root ]]; then
+    if (( !create )); then
+        printf >&2 '%s: %q is not a directory\n' "$argv0" "$directory"/root
```

Comment:
Don't use `%q` as it can encode the argument in a manner which will not be identical to what the user provided causing confusion.